packages/ios-client/Sources/FishjamClient/webrtc/PeerConnectionManager.swift

```diff
@@ -3,68 +3,187 @@ import Logging
+    private var streamIds: [String] = [UUID().uuidString]
 
-    // a common stream ID used for all non-screenshare and audio tracks
-    private let localStreamId = UUID().uuidString
+    internal init(config: RTCConfiguration, peerConnectionFactory: PeerConnectionFactoryWrapper) {
+        self.config = config
+        self.peerConnectionFactory = peerConnectionFactory
+    }
 
-    private let peerConnectionListener: PeerConnectionListener
+    private func getSendEncodingsFromSimulcastConfig(_ simulcastConfig: SimulcastConfig) -> [RTCRtpEncodingParameters] {
+        let sendEncodings = Constants.simulcastEncodings()
+        simulcastConfig.activeEncodings.forEach { encoding in
+            sendEncodings[encoding.rawValue].isActive = true
+        }
+        return sendEncodings
+    }
 
-    private var peerConnectionStats: [String: RTCStats] = [:]
+    public func addTrack(track: Track) {
+        addTrack(track: track, streamsId: streamIds)
+    }
 
-    internal init(
-        config: RTCConfiguration, peerConnectionFactory: PeerConnectionFactoryWrapper,
-        peerConnectionListener: PeerConnectionListener
-    ) {
-        self.config = config
-        iceServers = []
-        self.peerConnectionFactory = peerConnectionFactory
-        self.peerConnectionListener = peerConnectionListener
+    public func addTrack(track: Track, streamsId: [String]) {
+        guard let pc = connection else {
+            return
+        }
+        let videoParameters =
+            (track as? LocalVideoTrack)?.videoParameters ?? (track as? LocalScreencastTrack)?.videoParameters
+        let simulcastConfig = videoParameters?.simulcastConfig
+        var sendEncodings: [RTCRtpEncodingParameters] = []
+        if track.mediaTrack?.kind == "video"
+            && (track as? LocalVideoTrack)?.videoParameters.simulcastConfig.enabled == true
+        {
+            sendEncodings = getSendEncodingsFromSimulcastConfig(simulcastConfig!)
+        } else {
+            sendEncodings = [RTCRtpEncodingParameters.create(active: true)]
+        }
+
+        if let maxBandwidth = (track as? LocalVideoTrack)?.videoParameters.maxBandwidth {
+            applyBitrate(encodings: sendEncodings, maxBitrate: maxBandwidth)
+        }
+        let transceiverInit = RTCRtpTransceiverInit()
+        transceiverInit.direction = RTCRtpTransceiverDirection.sendOnly
+        transceiverInit.streamIds = streamsId
+        transceiverInit.sendEncodings = sendEncodings
+        pc.addTransceiver(with: track.mediaTrack!, init: transceiverInit)
+        pc.enforceSendOnlyDirection()
     }
 
-    public func close() {
-        if let pc = connection {
-            pc.close()
-            self.connection = nil
+    private func applyBitrate(encodings: [RTCRtpEncodingParameters], maxBitrate: TrackBandwidthLimit) {
+        switch maxBitrate {
+        case .BandwidthLimit(let limit):
+            splitBitrate(encodings: encodings, bitrate: limit)
+            break
```

Comment:
In Swift `break` is not required if you don't ignore the case, fallthrough is not implict: https://docs.swift.org/swift-book/documentation/the-swift-programming-language/controlflow/#No-Implicit-Fallthrough